src/hssm/hssm.py

```diff
@@ -598,8 +609,97 @@ def sample(
                 # if parent already in posterior
                 del self._inference_obj.posterior["rt,response_mean"]
 
+        # Subset data vars in posterior
+        if hasattr(self, "pymc_model") and self._inference_obj is not None:
+            vars_to_keep = set(
+                [var.name for var in getattr(self, "pymc_model").free_RVs]
+            ).intersection(set(list(self._inference_obj["posterior"].data_vars.keys())))
+        else:
+            raise ValueError("Model has not been initialized yet. Something is wrong!")
+
+        setattr(
+            self._inference_obj,
+            "posterior",
+            self._inference_obj["posterior"][list(vars_to_keep)],
+        )
+
         return self.traces
 
+    def vi(
+        self,
+        method: str = "advi",
+        niter: int = 10000,
+        draws=1000,
+        return_idata=True,
+        **vi_kwargs,
+    ) -> pm.Approximation | az.InferenceData:
+        """Perform Variational Inference.
+
+        Parameters
+        ----------
+        niter : int
```

Comment:
The docstring format is `numpy-notype`. Since we already use type annotation, we don't need to provide the types for the variables here. `mkdocs` can figure it out.